LastRunning-17_10_18

itaycsguy opened the pull request:
Check code before we are going to handle specific last things and android compilation ops.

itaycsguy committed:
init

itaycsguy committed:
Changed App theme view.
Done:
1. View,Theme and functionality on each page
2. Registration informative messages
3. added new class named CustomSnackBar for static snackBar usage
4. added new function in utils named hideKeyboard to hide the keyboard
5. added new class named PasswordManager to handle password strength issues
6. added real function using firebase engine for email sending under recovery page - bug: need to implement some callback method to retain the new password somewhere
7. checked the login flow well.
Things left:
1. add cancel button into EditText to let the user clear all input in once
2. find X icon to sign incorrect input [instead of the icon that is running at this moment]
3. Goto Morag's page and Progile page and change profile page to nice view as match as to morag's page view looks like.
4. put Profile page and Morag's page collapsed together into actionActivity form - 3 actions:
a. go to Morag's page
b. go to his profile
c. logout

itaycsguy committed:
Merge pull request #6 from itaycsguy/OctoberWork-Morag

Crop Updated and works with Upload and Loading spinner

itaycsguy committed:
1. Merged and compiled Morag's Activity with the new Crop version and excecuted well.
More logic is gathered from my last Branch such as: Action Bar [navigation between activities as comportable as it could be],informative noticed and so on.
2. It is runnning with some bugs: a. after changing fragment activity with another xml file, it is taken some part of the activity and draw it by white color - Morag's activity too!
3. next time destination is to done those issues.

itaycsguy committed:
1. Changed Menu management to use action bar to force and back from the camera operational page
2. changed sign in image

Morag140 commented on app/src/main/java/app/itaycsguy/musiciansaidb/AppAuth.kt:
```diff
@@ -57,22 +67,17 @@ class AppAuth(act : AppCompatActivity, fbDb : FirebaseDB) {
-                _signInResult = data.getSerializableExtra("data") as HashMap<String, String>
-            }
-        }
+        if(data!!.hasExtra("data")) _signInResult = data.getSerializableExtra("data") as HashMap<String, String>
```
since data is intent? , the "?" means it can be NULL and when u do data!! , the "!!" means it cannot be NULL...this is not good and can result in Null pointer exception.

the solution is the code u deleted which makes sure this code runs only if data exists (this is what .let() do )

Morag140 commented on app/src/main/java/app/itaycsguy/musiciansaidb/AppAuth.kt:
```diff
@@ -38,12 +48,12 @@ class AppAuth(act : AppCompatActivity, fbDb : FirebaseDB) {
 
     private fun checkExistAccount(email: String,password: String) {
-        (_firebassDB.getRef())?.child("users/${FirebaseDB.encodeUserEmail(email)}")?.ref?.addListenerForSingleValueEvent(object : ValueEventListener {
+        (_fbDb.getRef())!!.child("users/${FirebaseDB.encodeUserEmail(email)}").ref.addListenerForSingleValueEvent(object : ValueEventListener {
```
Look at comment below and return this to the deleted code as well

Morag140 commented on app/src/main/res/layout/activity_profile.xml:
```diff
@@ -6,174 +6,119 @@
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:background="@mipmap/background1"
```
i actually think this background color is better (its also the color i changed my menu activity to.
up to u but i think this should be our background.

Morag140 reviewed (commented):
After changes i asked for in the comments i see no reason not to merge this 👍 

* One more thing the back button u did looks like "close the app" button the "back" button i will add is better so i will replace it on my commit...leave it as it is for now till we put my button into the drawble

itaycsguy committed:
1. added more informative pop ups to user while he registers to application.
2. changed email regex and password method to check correctness.
3. changed some buttons from images to names and images to more informative images.
4. changed profile variety colors and style to match to camera operations page - need to discuss about some more.
